src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/dto/MigrateRequest.kt

```diff
@@ -0,0 +1,55 @@
+package uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.dto
+
+import io.swagger.v3.oas.annotations.media.Schema
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.SYSTEM_USERNAME
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.jpa.NonAssociation
+
+@Schema(description = "Migrate Request")
+data class MigrateRequest(
+  @Schema(description = "Prisoner number to not associate", required = true, example = "A1234BC")
+  val firstPrisonerNumber: String,
+  @Schema(
+    description = "Reason why this prisoner should be kept apart from the other",
+    required = true,
+    example = "VICTIM",
+  )
+  val firstPrisonerReason: NonAssociationReason,
+  @Schema(description = "Prisoner number to not associate", required = true, example = "D5678EF")
+  val secondPrisonerNumber: String,
+  @Schema(
+    description = "Reason why this prisoner should be kept apart from the other",
+    required = true,
+    example = "PERPETRATOR",
+  )
+  val secondPrisonerReason: NonAssociationReason,
+
+  @Schema(description = "Type of restriction, e.g. don't locate in the same cell", required = true, example = "CELL")
+  val restrictionType: NonAssociationRestrictionType,
+
+  @Schema(
+    description = "Type of restriction, e.g. don't locate in the same cell",
+    required = false,
+    example = "John and Luke always end up fighting",
+  )
+  val comment: String? = null,
+
+  @Schema(description = "Who authorised the non-association", required = false, example = "John Smith")
+  val authorisedBy: String? = null,
+) {
+  fun toNewEntity(): NonAssociation {
+    return NonAssociation(
+      id = null,
+      firstPrisonerNumber = firstPrisonerNumber,
+      firstPrisonerReason = firstPrisonerReason,
+      secondPrisonerNumber = secondPrisonerNumber,
+      secondPrisonerReason = secondPrisonerReason,
+      restrictionType = restrictionType,
+      comment = comment ?: "NO COMMENT",
```

Comment:
I'd probably leave as empty string as it's more neutral and potentially leave the UI the ability to detect this and possibly show whatever message content decide (and can be changed).

That would be my preference as it's more neutral, but not a biggie tho :)